opentracing-jdbc/opentracing-jdbc/src/main/java/org/zalando/opentracing/jdbc/span/PeerSpanDecorator.java

```diff
@@ -0,0 +1,73 @@
+package org.zalando.opentracing.jdbc.span;
+
+import io.opentracing.Span;
+import io.opentracing.tag.StringTag;
+import io.opentracing.tag.Tag;
+import org.apiguardian.api.API;
+
+import javax.annotation.Nullable;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import static io.opentracing.tag.Tags.PEER_HOSTNAME;
+import static io.opentracing.tag.Tags.PEER_HOST_IPV6;
+import static io.opentracing.tag.Tags.PEER_PORT;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+/**
+ * Sets the following span tags:
+ *
+ * <ul>
+ *     <li><code>peer.address</code></li>
+ *     <li><code>peer.hostname</code></li>
+ *     <li><code>peer.port</code></li>
+ * </ul>
+ *
+ * @see <a href="https://opentracing.io/specification/conventions/#span-tags-table">Standard Span Tags</a>
+ */
+@API(status = EXPERIMENTAL)
+public final class PeerSpanDecorator implements SpanDecorator {
+
+    private final PeerParser parser = PeerParser.create();
+
+    /**
+     * {@link io.opentracing.tag.Tags#PEER_HOST_IPV4} is typed as a {@link Tag Tag<Integer>} for some unknown reason.
+     * This version of the tag fixes that.
+     *
+     * @see io.opentracing.tag.Tags#PEER_HOST_IPV4
+     */
+    @API(status = INTERNAL)
+    private static final Tag<String> PEER_HOST_IPV4 = new StringTag("peer.ipv4");
+
+    /**
+     * Remote "address", suitable for use in a networking client library. This may be a "ip:port", a bare "hostname",
+     * a FQDN, or even a JDBC substring like "mysql://prod-db:3306".
+     *
+     * <strong>Be aware that this will be removed as soon as {@link io.opentracing.tag.Tags} contains it.</strong>
+     *
+     * @see <a href="https://github.com/opentracing/opentracing-java/pull/358">opentracing/opentracing-java#358</a>
+     */
+    @API(status = INTERNAL)
+    private static final Tag<String> PEER_ADDRESS = new StringTag("peer.address");
+
+    @Override
+    public void onQuery(final Span span, final Statement statement, final List<String> queries) throws SQLException {
+        final String url = statement.getConnection().getMetaData().getURL();
+        final Peer peer = parser.parse(url);
+
+        tag(span, PEER_ADDRESS, peer.getAddress());
```

Comment:
* see comment above on credentials
* also kinda does what `DatabaseInstance..` does

Reply:
I'll extract the tagging with `peer.address` into a `PeerAddressSpanDecorator` that is disabled by default.

Reply:
Done